examples/component/http-keyvalue-crud/main.go

```diff
@@ -0,0 +1,155 @@
+
+	// The wasmCloud wasihttp module enables us to write more idiomatic Go when using wasi:http.
+	"go.wasmcloud.dev/component/net/wasihttp"
+)
+
+// Types for JSON validation.
+type CheckRequest struct {
+	Value string `json:"value"`
+}
+
+type CheckResponse struct {
+	Valid   bool   `json:"valid"`
+	Length  int    `json:"length,omitempty"`
+	Message string `json:"message,omitempty"`
+}
+
+func init() {
+	// Establishes the routes and methods for our keyvalue operations.
+	router := httprouter.New()
+	router.GET("/", indexHandler)
+	router.POST("/crud/:key", postHandler)
+	router.GET("/crud/:key", getHandler)
+	router.DELETE("/crud/:key", deleteHandler)
+	wasihttp.Handle(router)
+}
+
+func indexHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	fmt.Fprintf(w, "GET, POST, or DELETE to /crud/<key> (with JSON payload for POSTs)\n")
+}
+
+func postHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+
+	// Assigns the "key" paramater to the "key" variable.
+	key := ps.ByName("key")
+
+	// Checks the request for a valid JSON body and assigns it to the value variable.
+	// The user will set the value via JSON payload:
+	// curl -X POST 'localhost:8000/crud/key' -d '{"foo": "bar", "woo": "hoo"}'
+	var req CheckRequest
+	defer r.Body.Close()
+	value, err := io.ReadAll(r.Body)
+	if err != nil {
+		errResponseJSON(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	if err := json.Unmarshal(value, &req); err != nil {
+		errResponseJSON(w, http.StatusBadRequest, fmt.Sprintf("error with json input: %s", err.Error()))
+		return
+	}
+
+	// Opens the keyvalue bucket.
+	kvStore := store.Open("default")
+	if err := kvStore.Err(); err != nil {
+		w.Write([]byte("Error: " + err.String()))
+		return
+	}
+
+	// Converts the value to a byte array.
+	valueBytes := []byte(value)
+
+	// Converts the byte array to the Component Model's cm.List type.
+	valueList := cm.ToList(valueBytes)
+
+	// Sets the value for the key in the current bucket and handles any errors.
+	kvSet := store.Bucket.Set(*kvStore.OK(), key, valueList)
+	if err := kvSet.Err(); err != nil {
+		w.Write([]byte("Error: " + err.String()))
+		return
+	} else {
+		// Confirms that the key has been set.
+		fmt.Fprintf(w, "Set %s to %s\n", key, value)
+	}
+}
+
+func getHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+
+	// Assigns the "key" paramater to the "key" variable.
+	key := ps.ByName("key")
+
+	// Opens the keyvalue bucket.
+	kvStore := store.Open("default")
+	if err := kvStore.Err(); err != nil {
+		w.Write([]byte("Error: " + err.String()))
+		return
+	}
+
+	// Gets the value for the defined key.
+	kvGet, kvGetErr, kvGetIsErr := store.Bucket.Get(*kvStore.OK(), key).Result()
+	if err := kvGetErr; kvGetIsErr {
+		w.Write([]byte("Error: " + err.String()))
+		return
+	} else if kvGet.Value().Len() == 0 {
+
+		// Reports that key does not exist if no value is found.
+		fmt.Fprintf(w, "%s does not exist\n", key)
+
```

Comment:
It may be better to write this as a separate early return if statement:
```go
    if kvGet.Value().Len() == 0 {
		// Reports that key does not exist if no value is found.
		fmt.Fprintf(w, "%s does not exist\n", key)
		return
    }
```